presto-main/src/main/java/com/facebook/presto/operator/aggregation/ChecksumAggregation.java

```diff
@@ -0,0 +1,131 @@
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Checksum of the given values";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type valueType = types.get("T");
+        Signature signature = new Signature(NAME, valueType.getTypeSignature(), valueType.getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(valueType);
+        return new FunctionInfo(signature, getDescription(), aggregation);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type type)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(ChecksumAggregation.class.getClassLoader());
+
+        List<Type> inputTypes = ImmutableList.of(type);
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, type, inputTypes),
+                createInputParameterMetadata(type),
+                INPUT_FUNCTION.bindTo(type),
+                null,
+                null,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                NullableLongState.class,
+                new StateCompiler().generateStateSerializer(NullableLongState.class, classLoader),
+                new StateCompiler().generateStateFactory(NullableLongState.class, classLoader),
+                VARBINARY,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, BIGINT, VARBINARY, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type type)
+    {
+        return ImmutableList.of(new ParameterMetadata(STATE), new ParameterMetadata(NULLABLE_BLOCK_INPUT_CHANNEL, type), new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(Type type, NullableLongState state, Block block, int position)
+    {
+        state.setNull(false);
+        if (block.isNull(position)) {
+            state.setLong(state.getLong() + 1);
+        }
+        else {
+            state.setLong(state.getLong() + type.hash(block, position));
+        }
+    }
+
+    public static void combine(NullableLongState state, NullableLongState otherState)
+    {
+        state.setNull(state.isNull() && otherState.isNull());
+        state.setLong(state.getLong() + otherState.getLong());
+    }
+
+    public static void output(NullableLongState state, BlockBuilder out)
+    {
+        if (state.isNull()) {
+            out.appendNull();
+        }
+        else {
+            VARBINARY.writeSlice(out, wrappedLongArray(state.getLong()));
```

Comment:
Both `Type.hash` and `HASH_CODE` operator produces 32-bit numbers. (former 32-bit int, latter 32-bit `BIGINT`)
Therefore, it will have a very bad distribution if output is 64-bit.